examples/simple/src/comments/PostReferenceInput.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogContent from '@material-ui/core/DialogContent';
+
+import { ReferenceInput, SelectInput } from 'react-admin'; // eslint-disable-line import/no-unresolved
+
+import PostQuickCreate from './PostQuickCreate';
+import { hidePostQuickCreate, showPostQuickCreate } from './postQuickCreate';
+
+class PostReferenceInputView extends React.Component {
+    static propTypes = {
+        hidePostQuickCreate: PropTypes.func.isRequired,
+        showDialog: PropTypes.bool,
+        showPostQuickCreate: PropTypes.func.isRequired,
+    };
+
+    handleClick = event => {
+        event.preventDefault();
+        this.props.showPostQuickCreate();
+    };
+
+    handleClose = () => {
+        this.props.hidePostQuickCreate();
+    };
+
+    render() {
+        const {
+            showDialog,
+            hidePostQuickCreate,
+            showPostQuickCreate,
+            ...props
+        } = this.props;
+
+        return (
+            <div>
+                <ReferenceInput {...props}>
+                    <SelectInput optionText="title" />
+                </ReferenceInput>
+                <Button onClick={this.handleClick}>New</Button>
+                <Dialog
+                    open={showDialog}
+                    onClose={this.handleClose}
+                    aria-label="New post"
+                >
+                    <DialogContent>
+                        <PostQuickCreate />
+                    </DialogContent>
+                </Dialog>
+            </div>
+        );
+    }
+}
+
+const PostReferenceInput = connect(
```

Comment:
isn't it the type of very local state that we'd better handle in pure React?

Reply:
The problem is, how do you know when to close the modal ? I wanted to close it only when the create succeded

Reply:
Then use optimistic updates ;)

Reply:
And how do we react to failures ?

Reply:
refresh and show an error notification (it's what we do for optimistic deletes)

Reply:
I do agree on one thing though: the UI could be improved. This is a very rough and naive exemple

Reply:
> refresh and show an error notification (it's what we do for optimistic deletes)

Even after the user has then submitted the Comment Create form ?